### ISSUE TITLE
KuCoin API update

### ISSUE THREAD

**Issue by mchalise** [SebastianJ/Kucoin#3]
I heard that KuCoin is upgrading their API. Does it hamper this gem? Could you please verify it and lemme know.

**Comment by mchalise** [comment]
@SebastianJ  kucoin platform upgrade done. FYI

**Comment by unplugandplay** [comment]
`https://api.kucoin.com/v1/open/currencies` should be `https://api.kucoin.com/api/v1`

**Comment by SebastianJ** [comment]
@unplugandplay Yeah I know, this was for API v1. I started working on an initial upgrade to version 2 today.

**Comment by SebastianJ** [comment]
Ok, I've taken an initial stab at upgrading the entire gem to work with KuCoin's new API.

There are still some endpoints (mainly transfers between accounts/sub-accounts and withdrawals) that haven't been implemented. Creating orders and fetching symbols, currencies etc. work.

Haven't cut a new gem release yet so use the repo for now and try it out when you guys have time.

Pull requests are welcome!

**Comment by unplugandplay** [comment]
Thx! 

I upgrade my fork tonight, test the brand NEW version, and try some pull request during this week ;)
